sdk/digitaltwins/Azure.DigitalTwins.Core/tests/NewtonsoftObjectSerializerTests.cs

```
@@ -0,0 +1,93 @@
+            string roomModelId = await GetUniqueModelIdAsync(defaultClient, TestAssetDefaults.RoomModelIdPrefix).ConfigureAwait(false);
+
+            try
+            {
+                // arrange
+
+                // create room model
+                string roomModel = TestAssetsHelper.GetRoomModelPayload(roomModelId, floorModelId);
+                await CreateAndListModelsAsync(defaultClient, new List<string> { roomModel }).ConfigureAwait(false);
+
+                // act
+
+                // create room twin
+                BasicDigitalTwin roomTwin = TestAssetsHelper.GetRoomTwinPayload(roomModelId);
+                await defaultClient.CreateOrReplaceDigitalTwinAsync<BasicDigitalTwin>(roomTwinId, roomTwin).ConfigureAwait(false);
+
+                // Create a client with NewtonsoftJsonObjectSerializer configured as the serializer.
+                DigitalTwinsClient testClient = GetClient(
+                    new DigitalTwinsClientOptions
+                    {
+                        Serializer = new NewtonsoftJsonObjectSerializer()
+                    });
+
+                // Get digital twin using the simple DigitalTwin model annotated with Newtonsoft attributes
+                SimpleNewtonsoftDtModel getResponse = await testClient.GetDigitalTwinAsync<SimpleNewtonsoftDtModel>(roomTwinId).ConfigureAwait(false);
+
+                Assert.IsNotNull(getResponse.Id, "Digital twin Id should not be null or empty");
+
+                // Query DigitalTwins using the simple DigitalTwin model annotated with Newtonsoft attributes
+                AsyncPageable<SimpleNewtonsoftDtModel> queryResponse = testClient.QueryAsync<SimpleNewtonsoftDtModel>("SELECT * FROM DIGITALTWINS");
+
+                await foreach (SimpleNewtonsoftDtModel twin in queryResponse)
+                {
+                    Assert.IsNotNull(twin.Id, "Digital twin Id should not be null or empty");
+                }
+            }
+            catch (Exception ex)
```

Why catch here? If `Assert.IsNotNull` fails on line 55 (for instance), it will throw, but this will catch and override the reason it failed.

---

Agreed, I will let it throw